catalog/pipeline/fbc-release/0.11/fbc-release.yaml

```diff
@@ -0,0 +1,221 @@
+      type: string
+      description: Max seconds to wait until the status is updated
+    - name: buildTimeoutSeconds
+      type: string
+      description: Max seconds to wait until the build finishes
+  workspaces:
+    - name: release-workspace
+  results:
+    - name: requestMessage
+      value: $(tasks.add-fbc-contribution-to-index-image.results.requestMessage)
+    - name: requestReason
+      value: $(tasks.add-fbc-contribution-to-index-image.results.requestReason)
+  tasks:
+    - name: verify-enterprise-contract
+      taskRef:
+        resolver: "bundles"
+        params:
+          - name: bundle
+            value: quay.io/hacbs-contract/ec-task-bundle:snapshot
+          - name: kind
+            value: task
+          - name: name
+            value: verify-enterprise-contract
+      params:
+        - name: IMAGES
+          value: $(params.snapshot)
+        - name: SSL_CERT_DIR
+          value: /var/run/secrets/kubernetes.io/serviceaccount
+        - name: POLICY_CONFIGURATION
+          value: $(params.enterpriseContractPolicy)
+        - name: STRICT
+          value: "1"
+    - name: add-fbc-contribution-to-index-image
+      workspaces:
+        - name: input
+          workspace: release-workspace
+      taskRef:
+        resolver: "bundles"
+        params:
+          - name: bundle
+            value: quay.io/hacbs-release/task-create-internal-request:0.5
+          - name: kind
+            value: task
+          - name: name
+            value: create-internal-request
+      params:
+        - name: pipelineRunName
+          value: $(context.pipelineRun.name)
+        - name: request
+          value: "iib"
+        - name: params
+          value:
+            - name: binaryImage
+              value: "$(params.binaryImage)"
+            - name: fromIndex
+              value: "$(params.fromIndex)"
+            - name: overwriteFromIndex
+              value: "$(params.overwriteFromIndex)"
+            - name: buildTags
+              value: "$(params.buildTags)"
+            - name: addArches
+              value: "$(params.addArches)"
+            - name: buildTimeoutSeconds
+              value: "$(params.buildTimeoutSeconds)"
+            - name: fbcFragment
+              value: '$(params.snapshot)'
+              jsonKey: ".components[0].containerImage"
+      runAfter:
+        - verify-enterprise-contract
+    - name: extract-requester-from-release
+      taskRef:
+        resolver: "bundles"
+        params:
+          - name: bundle
+            value: quay.io/hacbs-release/task-kubernetes-actions:0.2
+          - name: kind
+            value: task
+          - name: name
+            value: kubernetes-actions
+      params:
+        - name: script
+          value: |
+            set -x
+
+            NAMESPACE=$(echo $(params.release) | cut -d '/' -f 1)
+            NAME=$(echo $(params.release) | cut -d '/' -f 2)
+
+            AUTHOR=$(kubectl get release ${NAME} -n ${NAMESPACE} \
```

Comment:
@davidmogar @scoheb how do I setup the RBAC so whatever SA that runs this pipeline has the cluster role to get releases?

Reply:
I don't think you can do that. You always need to have a binding for the SA.

Reply:
Then how will we ever retrieve the cr that we are passing object references for (this isn't just for the release)? If we can't we will probably have to always pass json versions of them

Reply:
for the **InternalRequest** CR creation, I've had to manually create a RoleBinding for the service account.

I think we should create a new role - ReleaseServiceAccountRole that has what is needed for InternalRequests creation. listing and watching and that has what is needed for this.

Then as part of the setup of a ServiceAccount, we need to create the RoleBinding.

Reply:
Role or ClusterRole?

Reply:
So are these manual things we do and write down or they go as yamls in some repo or what? I know that the RBAC is needed for this to work, just not where to put it. Curious about what to do for the purposes of this ticket. If we aren't doing a ClusterRole, we will need one Role for each dev namespace. But we also don't know the SA name ahead of time (because it is a ReleaseStrategy param) so I am not sure how we would put the RoleBinding in code

Reply:
I guess it is safest to do **Role**?

Reply:
I think we'd need a Role defined by the Release Service that could exist in infra-deployment. We would manually create the RoleBinding when a Managed Workspace is created (automatically in e2e-tests)

Reply:
Will be handled by https://issues.redhat.com/browse/HACBS-2310
Leaving this thread open though until that ticket is complete